kod/util/factgame/territry.kod

```diff
@@ -396,18 +396,24 @@ messages:
       return;
    }
 
-   ReloadSavedFlagStates()
+   ReloadFlagStates(lFlagStates=$)
    {
       local i, iRID, oRoom, oFlag;
 
-      if plFlagStates = $
+      if lFlagStates = $
       {
-         DEBUG("ReloadSavedFlagStates:: nothing saved, nothing reloaded");
+         lFlagStates = plFlagStates;
       }
 
-      if length(plFlagStates) <> length(plFlagRIDs)
+      if lFlagStates = $
+         OR NOT IsList(lFlagStates)
       {
-         DEBUG("ReloadSavedFlagStates:: error in saved flag states - length mismatch");
+         DEBUG("ReloadFlagStates:: nothing saved, nothing reloaded");
+      }
+
+      if length(lFlagStates) <> length(plFlagRIDs)
+      {
+         DEBUG("ReloadFlagStates:: error in saved flag states - length mismatch");
```

Comment:
Probably just return here, since it's about to crash anyway (falling off the end of a list)?